myceli/tests/listener_test.rs

```diff
@@ -0,0 +1,146 @@
+mod utils;
+
+use messages::{ApplicationAPI, Message};
+use std::time::Duration;
+use tokio::time::sleep;
+use utils::{TestController, TestListener};
+
+#[tokio::test]
+pub async fn test_verify_listener_alive() {
+    let listener = TestListener::new();
+    listener.start().await.unwrap();
+
+    let mut controller = TestController::new().await;
+
+    let response = controller
+        .send_and_recv(&listener.listen_addr, Message::request_available_blocks())
+        .await;
+
+    assert_eq!(response, Message::available_blocks(vec![]));
+}
+
+#[tokio::test]
+pub async fn test_transmit_receive_block() {
```

Comment:
There is some duplication in these tests.  Not a big deal, but there could be an abstraction to reuse for them.

Reply:
Do you mean duplication in the setup for each test, or in the scenarios under test?